### DISCUSSION TITLE
Could FJT feedback msg allocation size be reduced?

### DISCUSSION THREAD

**Discussion by gavanderhoorn** [Yaskawa-Global/motoros2#98] [Q&A]
In the current implementation, the FJT action server configures the memory allocation for a feedback message as follows:

https://github.com/Yaskawa-Global/motoros2/blob/f26d997598108eff7d944335a052c5b6761039fa/src/ActionServer_FJT.c#L65-L71

IIRC and IIUC, `micro_ros_utilities_create_message_memory(..)` will take that configuration and allocate sufficient memory for it (we do that in a static memory block, but still).

I'm wondering whether that size could be reduced a bit, as I believe we would have access to the actual nr of axes and groups configured on the controller at that point.

For my controller fi, `MAX_CONTROLLABLE_GROUPS * MP_GRP_AXES_NUM == 8 * 8 == 64`, while the controller actually only has 7 axes (6 in `R1` and 1 in `B1`). As a [feedback message](https://github.com/ros-controls/control_msgs/blob/c9ce53dfb5e87b3dfd60bdd85b4826b67989ced4/control_msgs/action/FollowJointTrajectory.action#L52-L61) contains quite some fields, reducing these nrs could lead to a significant reduction in memory usage.

---

Edit: perhaps a more elaborate configuration struct should be used there, like what's created for goal message:

https://github.com/Yaskawa-Global/motoros2/blob/f26d997598108eff7d944335a052c5b6761039fa/src/ActionServer_FJT.c#L86-L105

?

**Comment by ted-miller** [comment]
> we do that in a static memory block, but still

Actually, the feedback msg is on the heap. It's the goal message that's in a static block.

So yes, I definitely agree that we should allocate more intelligently.
